addons/bus/static/src/multi_tab_service.js

```diff
@@ -157,12 +159,19 @@ export const multiTabService = {
             setItemInStorage("lastPresenceByTab", lastPresenceByTab);
 
             // Unload main.
-            if (_isOnMainTab) {
-                _isOnMainTab = false;
+            if (isOnMainTab()) {
                 bus.trigger("no_longer_main_tab");
                 browser.localStorage.removeItem(generateLocalStorageKey("main"));
             }
         }
+        /**
+         * Determine whether or not this tab is the main one.
+         *
+         * @returns {boolean}
+         */
+        function isOnMainTab() {
+            return getItemFromStorage("main") === tabId;
+        }
```

Comment:
Why remove `_isOnMainTab`?

`getItemFromStorage` is a disk read, so blocking and slow. The variable is meant to keep code fast.

Reply:
`_isOnMainTab` is unreliable, every tab is internally maintaining the state of this boolean variable with no knowledge of what the other tabs are doing. In other words: multiple tab may think they are the "main" at the same time. 

This approach is heavier but reliable, I'd rather wait for an answer I can trust than receive an instant response that may be incorrect.

Reply:
We could have hybrid version, where the read still is on variable but there's a local storage hook to observe change of this local storage item to update variable. That way the getter stays fast at all time

Reply:
So basically `onStorage` on this item updates the variable accordingly

Reply:
I messed around with your idea quite a bit and it works 90% of the time but the remaining 10% of recovery attempts fail because of concurrency issues.

The root of the problem is that the "onstorage" event is triggered when the local storage is updated **from another tab**.

Therefore, the process is the following:
- If I am the main tab, I update the local storage and then set internal state of `_isOnMainTab` to `true`.
- When `onstorage` event is fired, I know that another tab claimed to be main so, set internal state of `_isOnMainTab` to `false`.

Reminder, when the main tab is unloaded, every other active tabs will start the election simultaneously.

Result: In some cases, two tabs will claim to be the main tab at the same time and set their respective internal state to `true`. However, shortly after the `onstorage` event is fired... 
In the end, there is a `tabId` written in the local storage but the internal state of each tab is `_isOnMainTab = false`.